### PR TITLE
ETCM-389 Move checkpoint creation to the checkpoint service

### PR DESCRIPTION
**Pull request by AnastasiiaL**
# Description

Checkpoint creation moved to a respective service, so checkpoints and mined blocks are handled uniformly

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala [hunk 1]
```diff
@@ -69,39 +65,44 @@ class BlockImporter(
...
+        importBlock(
+          block,
+          new MinedBlockImportMessages(block),
+          MinedBlockImport,
+          false,
+          true)(state)
 
-    //TODO ETCM-389: Handle mined, checkpoint and new blocks uniformly
-    case nc @ NewCheckpoint(parentHash, signatures) =>
+    case nc @ NewCheckpoint(block) =>
```
Comment (dzajkowski) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala]: wouldn't it be more consistent to have:
```scala
case nc @ NewCheckpoint(block) if state.importing =>
context.system.scheduler.scheduleOnce(1.second, self, nc)
case nc @ NewCheckpoint(block) =>
...
```
?

Comment (AnastasiiaL) [reply src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala]: totally agree, missed this one

**Review by dzajkowski — APPROVED** [review]

File: src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala [hunk 2]
```diff
@@ -69,39 +65,33 @@ class BlockImporter(
       SignedTransaction.retrieveSendersInBackGround(blocks.toList.map(_.body))
       importBlocks(blocks, DefaultBlockImport)(state)
 
-    //TODO ETCM-389: Handle mined, checkpoint and new blocks uniformly
-    case MinedBlock(block) =>
-      if (!state.importing) {
-        importMinedBlock(block, state)
-      }
+    case MinedBlock(block) if !state.importing =>
+      importBlock(block, new MinedBlockImportMessages(block), MinedBlockImport, false, true)(state)
```
Comment (leo-bogastry) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala]: Can you please use named arguments, where you pass booleans? also in line 76

**Review by leo-bogastry — APPROVED** [review]
Nice to see the block handling simplified!

**Pull request merged**